[Bug] u8,u16,i8,i16 not working in struct

### Describe the bug

Bug for https://github.com/fzyzcjy/flutter_rust_bridge/discussions/428. The testcase repo is here: https://github.com/sagudev/u8u16.
Rust code:
```rust
pub struct Test {
    pub uu8: u8,
    pub uu16: u16,
    pub uu32: u32,
    pub uu64: u64,
    pub ii8: i8,
    pub ii16: i16,
    pub ii32: i32,
    pub ii64: i64,
}
```
The generated code is not working as apparently into_dart is not impl for u8/u16/i8/i16.
When running `flutter run` we get the following errors from cargo (rust): 
```
error[E0599]: the method `into_dart` exists for type `u8`, but its trait bounds were not satisfied
   --> src/bridge_generated.rs:116:22
    |
116 |             self.uu8.into_dart(),
    |                      ^^^^^^^^^ method cannot be called on `u8` due to unsatisfied trait bounds
    |
    = note: the following trait bounds were not satisfied:
            `u8: Into<DartCObject>`
            which is required by `u8: IntoDart`
            `&u8: Into<DartCObject>`
            which is required by `&u8: IntoDart`
            `&mut u8: Into<DartCObject>`
            which is required by `&mut u8: IntoDart`

error[E0599]: the method `into_dart` exists for type `u16`, but its trait bounds were not satisfied
   --> src/bridge_generated.rs:117:23
    |
117 |             self.uu16.into_dart(),
    |                       ^^^^^^^^^ method cannot be called on `u16` due to unsatisfied trait bounds
    |
    = note: the following trait bounds were not satisfied:
            `u16: Into<DartCObject>`
            which is required by `u16: IntoDart`
            `&u16: Into<DartCObject>`
            which is required by `&u16: IntoDart`
            `&mut u16: Into<DartCObject>`
            which is required by `&mut u16: IntoDart`

error[E0599]: the method `into_dart` exists for type `i8`, but its trait bounds were not satisfied
   --> src/bridge_generated.rs:120:22
    |
120 |             self.ii8.into_dart(),
    |                      ^^^^^^^^^ method cannot be called on `i8` due to unsatisfied trait bounds
    |
    = note: the following trait bounds were not satisfied:
            `i8: Into<DartCObject>`
            which is required by `i8: IntoDart`
            `&i8: Into<DartCObject>`
            which is required by `&i8: IntoDart`
            `&mut i8: Into<DartCObject>`
            which is required by `&mut i8: IntoDart`

error[E0599]: the method `into_dart` exists for type `i16`, but its trait bounds were not satisfied
   --> src/bridge_generated.rs:121:23
    |
121 |             self.ii16.into_dart(),
    |                       ^^^^^^^^^ method cannot be called on `i16` due to unsatisfied trait bounds
    |
    = note: the following trait bounds were not satisfied:
            `i16: Into<DartCObject>`
            which is required by `i16: IntoDart`
            `&i16: Into<DartCObject>`
            which is required by `&i16: IntoDart`
            `&mut i16: Into<DartCObject>`
            which is required by `&mut i16: IntoDart`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `native` due to 4 previous errors
Building Linux application...                                           
Exception: Build process failed
```

### Codegen logs with `RUST_LOG=debug` env variable

```shell
$ RUST_LOG=debug flutter_rust_bridge_codegen --llvm-path /usr/lib/llvm-14 --rust-input native/src/bridge_in.rs \
        --dart-output lib/bridge_generated.dart
[2022-04-29T10:05:56Z DEBUG flutter_rust_bridge_codegen::commands] execute command: bin=sh args=["-c", "dart pub global list"] current_dir=None cmd="sh" "-c" "dart pub global list"
[2022-04-29T10:05:56Z DEBUG flutter_rust_bridge_codegen::commands] command="sh" "-c" "dart pub global list" stdout=ffigen 4.1.3
     stderr=
[2022-04-29T10:05:56Z DEBUG flutter_rust_bridge_codegen::commands] execute command: bin=sh args=["-c", "test -x \"$(which cbindgen)\""] current_dir=None cmd="sh" "-c" "test -x \"$(which cbindgen)\""
[2022-04-29T10:05:56Z DEBUG flutter_rust_bridge_codegen::commands] command="sh" "-c" "test -x \"$(which cbindgen)\"" stdout= stderr=
[2022-04-29T10:05:56Z DEBUG flutter_rust_bridge_codegen::commands] 
[2022-04-29T10:05:56Z INFO  flutter_rust_bridge_codegen] Picked config: Opts { rust_input_path: "/home/samo/Namizje/u8u16/native/src/bridge_in.rs", dart_output_path: "/home/samo/Namizje/u8u16/lib/bridge_generated.dart", dart_decl_output_path: None, c_output_path: ["/tmp/.tmp1iLPvf.h"], rust_crate_dir: "/home/samo/Namizje/u8u16/native", rust_output_path: "/home/samo/Namizje/u8u16/native/src/bridge_generated.rs", class_name: "Native", dart_format_line_length: 80, skip_add_mod_to_lib: false, llvm_path: ["/usr/lib/llvm-14"], llvm_compiler_opts: "", manifest_path: "/home/samo/Namizje/u8u16/native/Cargo.toml", dart_root: Some("/home/samo/Namizje/u8u16"), build_runner: true }
[2022-04-29T10:05:56Z INFO  flutter_rust_bridge_codegen] Phase: Parse source code to AST
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', /home/samo/.cargo/registry/src/github.com-1ecc6299db9ec823/flutter_rust_bridge_codegen-1.28.0/src/main.rs:44:75
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
samo@medion:~/Namizje/u8u16$ RUST_LOG=debug flutter_rust_bridge_codegen --llvm-path /usr/lib/llvm-14 --rust-input ./native/src/bridge_in.rs         --dart-output lib/bridge_generated.dart
[2022-04-29T10:06:24Z DEBUG flutter_rust_bridge_codegen::commands] execute command: bin=sh args=["-c", "dart pub global list"] current_dir=None cmd="sh" "-c" "dart pub global list"
[2022-04-29T10:06:24Z DEBUG flutter_rust_bridge_codegen::commands] command="sh" "-c" "dart pub global list" stdout=ffigen 4.1.3
     stderr=
[2022-04-29T10:06:24Z DEBUG flutter_rust_bridge_codegen::commands] execute command: bin=sh args=["-c", "test -x \"$(which cbindgen)\""] current_dir=None cmd="sh" "-c" "test -x \"$(which cbindgen)\""
[2022-04-29T10:06:24Z DEBUG flutter_rust_bridge_codegen::commands] command="sh" "-c" "test -x \"$(which cbindgen)\"" stdout= stderr=
[2022-04-29T10:06:24Z DEBUG flutter_rust_bridge_codegen::commands] 
[2022-04-29T10:06:24Z INFO  flutter_rust_bridge_codegen] Picked config: Opts { rust_input_path: "/home/samo/Namizje/u8u16/./native/src/bridge_in.rs", dart_output_path: "/home/samo/Namizje/u8u16/lib/bridge_generated.dart", dart_decl_output_path: None, c_output_path: ["/tmp/.tmpUTTTJV.h"], rust_crate_dir: "/home/samo/Namizje/u8u16/./native", rust_output_path: "/home/samo/Namizje/u8u16/./native/src/bridge_generated.rs", class_name: "Native", dart_format_line_length: 80, skip_add_mod_to_lib: false, llvm_path: ["/usr/lib/llvm-14"], llvm_compiler_opts: "", manifest_path: "/home/samo/Namizje/u8u16/./native/Cargo.toml", dart_root: Some("/home/samo/Namizje/u8u16"), build_runner: true }
[2022-04-29T10:06:24Z INFO  flutter_rust_bridge_codegen] Phase: Parse source code to AST
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', /home/samo/.cargo/registry/src/github.com-1ecc6299db9ec823/flutter_rust_bridge_codegen-1.28.0/src/main.rs:44:75
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
samo@medion:~/Namizje/u8u16$ RUST_LOG=debug flutter_rust_bridge_codegen --llvm-path /usr/lib/llvm-14 --rust-input ./native/src/api.rs         --dart-output lib/bridge_generated.dart
[2022-04-29T10:06:43Z DEBUG flutter_rust_bridge_codegen::commands] execute command: bin=sh args=["-c", "dart pub global list"] current_dir=None cmd="sh" "-c" "dart pub global list"
[2022-04-29T10:06:44Z DEBUG flutter_rust_bridge_codegen::commands] command="sh" "-c" "dart pub global list" stdout=ffigen 4.1.3
     stderr=
[2022-04-29T10:06:44Z DEBUG flutter_rust_bridge_codegen::commands] execute command: bin=sh args=["-c", "test -x \"$(which cbindgen)\""] current_dir=None cmd="sh" "-c" "test -x \"$(which cbindgen)\""
[2022-04-29T10:06:44Z DEBUG flutter_rust_bridge_codegen::commands] command="sh" "-c" "test -x \"$(which cbindgen)\"" stdout= stderr=
[2022-04-29T10:06:44Z DEBUG flutter_rust_bridge_codegen::commands] 
[2022-04-29T10:06:44Z INFO  flutter_rust_bridge_codegen] Picked config: Opts { rust_input_path: "/home/samo/Namizje/u8u16/./native/src/api.rs", dart_output_path: "/home/samo/Namizje/u8u16/lib/bridge_generated.dart", dart_decl_output_path: None, c_output_path: ["/tmp/.tmpPTHbnM.h"], rust_crate_dir: "/home/samo/Namizje/u8u16/./native", rust_output_path: "/home/samo/Namizje/u8u16/./native/src/bridge_generated.rs", class_name: "Native", dart_format_line_length: 80, skip_add_mod_to_lib: false, llvm_path: ["/usr/lib/llvm-14"], llvm_compiler_opts: "", manifest_path: "/home/samo/Namizje/u8u16/./native/Cargo.toml", dart_root: Some("/home/samo/Namizje/u8u16"), build_runner: true }
[2022-04-29T10:06:44Z INFO  flutter_rust_bridge_codegen] Phase: Parse source code to AST
[2022-04-29T10:06:44Z INFO  flutter_rust_bridge_codegen] Phase: Parse AST to IR
[2022-04-29T10:06:44Z DEBUG flutter_rust_bridge_codegen::source_graph] Trying to parse "/home/samo/Namizje/u8u16/native/src/api.rs"
[2022-04-29T10:06:44Z DEBUG flutter_rust_bridge_codegen::source_graph] Trying to parse "/home/samo/Namizje/u8u16/native/src/bridge_generated.rs"
[2022-04-29T10:06:44Z DEBUG flutter_rust_bridge_codegen::parser] parse_function function name: Ident(platform)
[2022-04-29T10:06:44Z DEBUG flutter_rust_bridge_codegen::parser] parse_function function name: Ident(rust_release_mode)
[2022-04-29T10:06:44Z DEBUG flutter_rust_bridge_codegen::parser] parse_function function name: Ident(test)
[2022-04-29T10:06:44Z DEBUG flutter_rust_bridge_codegen] parsed functions: IrFile { funcs: [IrFunc { name: "platform", inputs: [], output: EnumRef(IrTypeEnumRef { name: "Platform", is_struct: false }), fallible: false, mode: Normal, comments: [] }, IrFunc { name: "rust_release_mode", inputs: [], output: Primitive(Bool), fallible: false, mode: Normal, comments: [] }, IrFunc { name: "test", inputs: [], output: StructRef(IrTypeStructRef { name: "Test" }), fallible: false, mode: Normal, comments: [] }], struct_pool: {"Test": IrStruct { name: "Test", wrapper_name: None, path: Some(["crate", "api", "Test"]), fields: [IrField { ty: Primitive(U8), name: IrIdent { raw: "uu8" }, comments: [] }, IrField { ty: Primitive(U16), name: IrIdent { raw: "uu16" }, comments: [] }, IrField { ty: Primitive(U32), name: IrIdent { raw: "uu32" }, comments: [] }, IrField { ty: Primitive(U64), name: IrIdent { raw: "uu64" }, comments: [] }, IrField { ty: Primitive(I8), name: IrIdent { raw: "ii8" }, comments: [] }, IrField { ty: Primitive(I16), name: IrIdent { raw: "ii16" }, comments: [] }, IrField { ty: Primitive(I32), name: IrIdent { raw: "ii32" }, comments: [] }, IrField { ty: Primitive(I64), name: IrIdent { raw: "ii64" }, comments: [] }], is_fields_named: true, comments: [] }}, enum_pool: {"Platform": IrEnum { name: "Platform", wrapper_name: None, path: ["crate", "api", "Platform"], comments: [], _variants: [IrVariant { name: IrIdent { raw: "Unknown" }, comments: [], kind: Value }, IrVariant { name: IrIdent { raw: "Android" }, comments: [], kind: Value }, IrVariant { name: IrIdent { raw: "Ios" }, comments: [], kind: Value }, IrVariant { name: IrIdent { raw: "Windows" }, comments: [], kind: Value }, IrVariant { name: IrIdent { raw: "Unix" }, comments: [], kind: Value }, IrVariant { name: IrIdent { raw: "MacIntel" }, comments: [], kind: Value }, IrVariant { name: IrIdent { raw: "MacApple" }, comments: [], kind: Value }, IrVariant { name: IrIdent { raw: "Wasm" }, comments: [], kind: Value }], _is_struct: false }}, has_executor: false }
[2022-04-29T10:06:44Z INFO  flutter_rust_bridge_codegen] Phase: Transform IR
[2022-04-29T10:06:44Z DEBUG flutter_rust_bridge_codegen] transformed functions: IrFile { funcs: [IrFunc { name: "platform", inputs: [], output: EnumRef(IrTypeEnumRef { name: "Platform", is_struct: false }), fallible: false, mode: Normal, comments: [] }, IrFunc { name: "rust_release_mode", inputs: [], output: Primitive(Bool), fallible: false, mode: Normal, comments: [] }, IrFunc { name: "test", inputs: [], output: StructRef(IrTypeStructRef { name: "Test" }), fallible: false, mode: Normal, comments: [] }], struct_pool: {"Test": IrStruct { name: "Test", wrapper_name: None, path: Some(["crate", "api", "Test"]), fields: [IrField { ty: Primitive(U8), name: IrIdent { raw: "uu8" }, comments: [] }, IrField { ty: Primitive(U16), name: IrIdent { raw: "uu16" }, comments: [] }, IrField { ty: Primitive(U32), name: IrIdent { raw: "uu32" }, comments: [] }, IrField { ty: Primitive(U64), name: IrIdent { raw: "uu64" }, comments: [] }, IrField { ty: Primitive(I8), name: IrIdent { raw: "ii8" }, comments: [] }, IrField { ty: Primitive(I16), name: IrIdent { raw: "ii16" }, comments: [] }, IrField { ty: Primitive(I32), name: IrIdent { raw: "ii32" }, comments: [] }, IrField { ty: Primitive(I64), name: IrIdent { raw: "ii64" }, comments: [] }], is_fields_named: true, comments: [] }}, enum_pool: {"Platform": IrEnum { name: "Platform", wrapper_name: None, path: ["crate", "api", "Platform"], comments: [], _variants: [IrVariant { name: IrIdent { raw: "Unknown" }, comments: [], kind: Value }, IrVariant { name: IrIdent { raw: "Android" }, comments: [], kind: Value }, IrVariant { name: IrIdent { raw: "Ios" }, comments: [], kind: Value }, IrVariant { name: IrIdent { raw: "Windows" }, comments: [], kind: Value }, IrVariant { name: IrIdent { raw: "Unix" }, comments: [], kind: Value }, IrVariant { name: IrIdent { raw: "MacIntel" }, comments: [], kind: Value }, IrVariant { name: IrIdent { raw: "MacApple" }, comments: [], kind: Value }, IrVariant { name: IrIdent { raw: "Wasm" }, comments: [], kind: Value }], _is_struct: false }}, has_executor: false }
[2022-04-29T10:06:44Z INFO  flutter_rust_bridge_codegen] Phase: Generate Rust code
[2022-04-29T10:06:44Z INFO  flutter_rust_bridge_codegen] Phase: Generate Dart code
[2022-04-29T10:06:44Z DEBUG flutter_rust_bridge_codegen::generator::dart] distinct_input_types=[]
[2022-04-29T10:06:44Z DEBUG flutter_rust_bridge_codegen::generator::dart] distinct_output_types=[Primitive(Bool), Primitive(I16), Primitive(I32), Primitive(I64), Primitive(I8), EnumRef(IrTypeEnumRef { name: "Platform", is_struct: false }), StructRef(IrTypeStructRef { name: "Test" }), Primitive(U16), Primitive(U32), Primitive(U64), Primitive(U8)]
[2022-04-29T10:06:44Z INFO  flutter_rust_bridge_codegen] Phase: Other things
[2022-04-29T10:06:44Z DEBUG flutter_rust_bridge_codegen::commands] execute format_rust path=/home/samo/Namizje/u8u16/./native/src/bridge_generated.rs
[2022-04-29T10:06:44Z DEBUG flutter_rust_bridge_codegen::commands] execute command: bin=rustfmt args=["/home/samo/Namizje/u8u16/./native/src/bridge_generated.rs"] current_dir=None cmd="rustfmt" "/home/samo/Namizje/u8u16/./native/src/bridge_generated.rs"
[2022-04-29T10:06:44Z DEBUG flutter_rust_bridge_codegen::commands] command="rustfmt" "/home/samo/Namizje/u8u16/./native/src/bridge_generated.rs" stdout= stderr=
[2022-04-29T10:06:44Z DEBUG flutter_rust_bridge_codegen::commands] execute cbindgen rust_crate_dir=/home/samo/Namizje/u8u16/./native c_output_path=/tmp/.tmpfAnKHu.h
[2022-04-29T10:06:44Z DEBUG flutter_rust_bridge_codegen::commands] cbindgen config: 
    language = "C"
    
    # do NOT include "stdarg.h", see #108 and #53
    sys_includes = ["stdbool.h", "stdint.h", "stdlib.h"]
    no_includes = true
    
    [export]
    include = ["wire_Test"]
    
[2022-04-29T10:06:44Z DEBUG flutter_rust_bridge_codegen::commands] cbindgen config_file: NamedTempFile("/tmp/.tmp2Vuvy8")
[2022-04-29T10:06:44Z DEBUG flutter_rust_bridge_codegen::commands] execute command: bin=cbindgen args=["-v", "--config", "/tmp/.tmp2Vuvy8", "--output", "/tmp/.tmpfAnKHu.h"] current_dir=Some("/home/samo/Namizje/u8u16/native") cmd="cbindgen" "-v" "--config" "/tmp/.tmp2Vuvy8" "--output" "/tmp/.tmpfAnKHu.h"
[2022-04-29T10:06:44Z DEBUG flutter_rust_bridge_codegen::commands] command="cbindgen" "-v" "--config" "/tmp/.tmp2Vuvy8" "--output" "/tmp/.tmpfAnKHu.h" stdout= stderr=INFO: Take native::Platform - opaque (Enum is not marked with a valid #[repr(prim)] or #[repr(C)].).
    INFO: Take native::Test - opaque (Struct is not marked #[repr(C)] or #[repr(transparent)].).
    INFO: Take native::wire_platform.
    INFO: Take native::wire_rust_release_mode.
    INFO: Take native::wire_test.
    INFO: Take native::free_WireSyncReturnStruct.
    INFO: Take native::DartPort.
    INFO: Take native::DartPostCObjectFnType.
    INFO: Take native::store_dart_post_cobject.
    INFO: Take native::WireSyncReturnStruct.
    
[2022-04-29T10:06:44Z DEBUG flutter_rust_bridge_codegen::commands] execute ffigen c_path=/tmp/.tmpfAnKHu.h dart_path=/tmp/.tmpNf9G4F llvm_path=["/usr/lib/llvm-14"]
[2022-04-29T10:06:44Z DEBUG flutter_rust_bridge_codegen::commands] ffigen config: 
            output: '/tmp/.tmpNf9G4F'
            name: 'NativeWire'
            description: 'generated by flutter_rust_bridge'
            headers:
              entry-points:
                - '/tmp/.tmpfAnKHu.h'
              include-directives:
                - '/tmp/.tmpfAnKHu.h'
            comments: false
            preamble: |
              // ignore_for_file: camel_case_types, non_constant_identifier_names, avoid_positional_boolean_parameters, annotate_overrides, constant_identifier_names
            
            llvm-path:
               - '/usr/lib/llvm-14'
    
[2022-04-29T10:06:44Z DEBUG flutter_rust_bridge_codegen::commands] ffigen config_file: NamedTempFile("/tmp/.tmpEeqFh8")
[2022-04-29T10:06:44Z DEBUG flutter_rust_bridge_codegen::commands] execute command: bin=sh args=["-c", "dart pub global run ffigen --config \"/tmp/.tmpEeqFh8\""] current_dir=None cmd="sh" "-c" "dart pub global run ffigen --config \"/tmp/.tmpEeqFh8\""
[2022-04-29T10:06:45Z DEBUG flutter_rust_bridge_codegen::commands] command="sh" "-c" "dart pub global run ffigen --config \"/tmp/.tmpEeqFh8\"" stdout=Running in Directory: '/home/samo/Namizje/u8u16'
    Input Headers: [/tmp/.tmpfAnKHu.h]
    Finished, Bindings generated in /tmp/.tmpNf9G4F
     stderr=
[2022-04-29T10:06:45Z DEBUG flutter_rust_bridge_codegen::commands] execute format_dart path=/home/samo/Namizje/u8u16/lib/bridge_generated.dart line_length=80
[2022-04-29T10:06:45Z DEBUG flutter_rust_bridge_codegen::commands] execute command: bin=sh args=["-c", "dart format /home/samo/Namizje/u8u16/lib/bridge_generated.dart --line-length 80"] current_dir=None cmd="sh" "-c" "dart format /home/samo/Namizje/u8u16/lib/bridge_generated.dart --line-length 80"
[2022-04-29T10:06:45Z DEBUG flutter_rust_bridge_codegen::commands] command="sh" "-c" "dart format /home/samo/Namizje/u8u16/lib/bridge_generated.dart --line-length 80" stdout=Formatted /home/samo/Namizje/u8u16/lib/bridge_generated.dart
    Formatted 1 file (1 changed) in 0.22 seconds.
     stderr=
[2022-04-29T10:06:45Z INFO  flutter_rust_bridge_codegen] Success! Now go and use it :)
```


### To Reproduce

_No response_

### Expected behavior

_No response_

### Generated binding code

_No response_

### OS

Linux

### Version of `flutter_rust_bridge_codegen`

1.28

### Flutter info

_No response_

### Version of `clang++`

_No response_

### Version of `cbindgen`

_No response_

### Version of `ffigen`

_No response_

### Additional context

_No response_

--- a/frb_rust/Cargo.toml
+++ b/frb_rust/Cargo.toml
@@ -9,7 +9,7 @@ keywords = ["flutter", "dart", "ffi", "code-generation", "bindings"]
 categories = ["development-tools::ffi"]
 
 [dependencies]
-allo-isolate = "0.1.12"
+allo-isolate = "0.1.13-beta.3"
 # allo-isolate = { git = "https://github.com/Desdaemon/allo-isolate" } # hack, until upstream PR is merged and released
 # allo-isolate = { path = "../../allo-isolate" }
 anyhow = "1.0.57"

